dynamiqs/time_qarray.py

```diff
@@ -314,10 +330,36 @@ def in_axes(self) -> PyTree[int | None]:
         pass
 
     @property
-    @abstractmethod
     def discontinuity_ts(self) -> Array:
         # must be sorted, not necessarily unique values
-        pass
+        times = []
+        if self.tstart is not None:
+            times += [self.tstart]
+        if self.tend is not None:
+            times += [self.tend]
+        return jnp.array(times) if len(times) > 0 else jnp.empty(0)
+
+    def clip(
+        self, tstart: float | None = None, tend: float | None = None
+    ) -> TimeQArray:
```

Comment:
I find the API and the warning a bit misleading: it's hard to understand exactly what's done under the hood. For instance, if one calls `x.clip(0, 1)` and then `x.clip(0.5)`, it's not necessarily evident that only `(0.5, None)` is kept.

I would thus propose to remove default values, which would at least force users to write `x.clip(0.5, None)`, and to update the warning to saying that if `None` is specified, it will override any existing value.

```suggestion
    def clip(self, tstart: float | None, tend: float | None) -> TimeQArray:
```